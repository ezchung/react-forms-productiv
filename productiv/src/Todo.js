import React from "react";

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/

function Todo({id, title, description, priority}) {
  console.log("title--->", title)
  console.log("desc--->", description)
  console.log("priority--->", priority)
  return (
      <div className="Todo">
        <div><b>{title}</b> <small>(priority: {priority})</small></div>
        <div><small>{description}</small></div>
      </div>
  );
}
//TODO:

export default Todo;