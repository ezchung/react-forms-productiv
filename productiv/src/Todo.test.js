import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";


/*********** Render and Smoke Tests ********/
it("renders without crashing", function() {
  render(<Todo
    id={1}
    title="test"
    description="test description"
    priority={1} />);
})

it("matches snapshot", function() {
  const {container} = render(<Todo
    id={1}
    title="test"
    description="test description"
    priority={1} />);
  expect(container).toMatchSnapshot();
});

/*********** Unit Test(s) ********/

/*Test that rendered to-do displays properly */
it("displays the todo data properly", function() {
  const {container, debug} = render(<Todo
    id={2}
    title="test2"
    description="test description2"
    priority={2} />);
  expect(container).toContainHTML("test2")
  expect(container).toContainHTML("test description2")
  expect(container).toContainHTML("(priority: 2)")
})