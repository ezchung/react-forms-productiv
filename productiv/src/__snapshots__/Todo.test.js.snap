// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        test
      </b>
       
      <small>
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test description
      </small>
    </div>
  </div>
</div>
`;
